nodebalancer_configs.go

```diff
@@ -103,6 +103,33 @@ type NodeBalancerConfigCreateOptions struct {
 	SSLKey        string           `json:"ssl_key,omitempty"`
 }
 
+// NodeBalancerConfigRebuildOptionsNode for a nodebalancer instance mapping
+type NodeBalancerConfigRebuildOptionsNode struct {
+	Address *string `json:"address"`
+	Label   *string `json:"label"`
+	Weight  *int    `json:"weight"`
+	Mode    *string `json:"mode"`
+}
+
+// NodeBalancerConfigRebuildOptions used by RebuildNodeBalancerConfig
+type NodeBalancerConfigRebuildOptions struct {
+	Port          int                                     `json:"port"`
+	Protocol      ConfigProtocol                          `json:"protocol,omitempty"`
+	Algorithm     ConfigAlgorithm                         `json:"algorithm,omitempty"`
+	Stickiness    ConfigStickiness                        `json:"stickiness,omitempty"`
+	Check         ConfigCheck                             `json:"check,omitempty"`
+	CheckInterval int                                     `json:"check_interval,omitempty"`
+	CheckAttempts int                                     `json:"check_attempts,omitempty"`
+	CheckPath     string                                  `json:"check_path,omitempty"`
+	CheckBody     string                                  `json:"check_body,omitempty"`
+	CheckPassive  *bool                                   `json:"check_passive,omitempty"`
+	CheckTimeout  int                                     `json:"check_timeout,omitempty"`
+	CipherSuite   ConfigCipher                            `json:"cipher_suite,omitempty"`
+	SSLCert       string                                  `json:"ssl_cert,omitempty"`
+	SSLKey        string                                  `json:"ssl_key,omitempty"`
+	Nodes         []*NodeBalancerConfigRebuildOptionsNode `json:"nodes,omitempty"`
```

Comment:
I'm not sure what the behavior is for a rebuild with an empty node array (`"nodes":[]`).   

If it removes all of the nodes (any differently than not specifying `"nodes"` in the json) we may want to avoid `omitempty` to allow one to clear the nodes.

---

Likewise, `CheckInterval`, `CheckTimeout`, `CheckAttempts`, `Port` may have meaningful `0` values to consider.
Likewise, `CheckPath`, `CheckBody`, may have special meaning as `""`.

This probably lines up with NodeBalancerConfigCreateOptions, and if so I would hope we could just alias the type.